### ISSUE TITLE
Can't have tag and the name of the post same

### ISSUE THREAD

**Issue by k8s-dev** [knadh/hugo-ink#37]
https://github.com/knadh/hugo-ink/pull/20 breaks the rendering if I have a post with the same name as of the tag.

example:

I have a post named background.md and on another post I have a tag as 'background', I get into error such that:

```
❯ hugo server -D
Start building sites …
Built in 26 ms
Error: Error building site: failed to render pages: render of "term" failed: "workspace/blog/themes/hugo-ink/layouts/_default/term.html:11:16": execute of template failed: template: _default/term.html:11:16: executing "_default/term.html" at <.Site.GetPage>: error calling GetPage: page reference "/background" is ambiguous
```

**Comment by mr-karan** [comment]
@k8s-dev This issue seems to have been fixed in v0.65 as per https://github.com/gohugoio/hugo/issues/6034. Which version are you at?

**Comment by k8s-dev** [comment]
```
hugo version                                                                                                          ─╯
Hugo Static Site Generator v0.79.1/extended darwin/amd64 BuildDate: unknown
```
